Skynode: Add docs for using vanilla PX4

hamishwillee opened the pull request:
Companion docs for https://github.com/PX4/PX4-Autopilot/pull/21842

This adds docs for Skynode, in order to explain how you build and upload vanilla PX4, and when/why you might do this. This is a special case in that you would only do this during development. So it is linked in the Companion computer docs, but in a separate section from other options. It does not attempt to be a comprehensive entry point but instead points through to Auterion docs.

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,72 @@
+
+Skynode PX4 `px4_fmu-v5x` binaries are built using the normal [developer environment](../dev_setup/dev_env.md) and are uploaded using either  `upload_skynode_usb` or `upload_skynode_wifi` upload targets.
+
+You must have already installed [SSH keys](https://docs.auterion.com/manufacturers/software-customization/adding-ssh-keys) and you will need to enter Skynode's password (twice) during the upload process.
```
Is this the root password: https://docs.auterion.com/manufacturers/software-customization/changing-the-root-password

If not, what link to Auterion docs can I use?

MaEtUgR replied:
That's the password and how to change it yes.

hamishwillee replied:
Ah. Note, this has to change. The development image is what can/should be modified, and that has fixed password etc.

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,72 @@
+`upload_skynode_usb` and `upload_skynode_wifi` connect to Skynode via SSH over a network interface using the default (fixed) IP addresses, and upload a TAR compressed binary to the mission computer.
+The mission computer then decompresses the binary and installs it to the flight controller.
+
+SSH and TAR are upload dependencies, but are expected to be present by default on Ubuntu, Ubuntu running on Windows in WSL2, and macOS.
```
Note, this is at the end, because these are expected to be present. If something isn't on macOS we can add a link when someone complains.

hamishwillee replied:
Thank you:

```suggestion
SSH and TAR are upload dependencies, but are expected to be present by default on Ubuntu and Ubuntu running on Windows in WSL2.
On macOS you should first install [gnu-tar](https://formulae.brew.sh/formula/gnu-tar).
```

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,72 @@
+
+## Restoring Auterion OS
+
+To restore Auterion OS:
```
How do we do this? Is there a link?

MaEtUgR replied:
I'm informing myself what the easiest way to revert to Auterion PX4 is. AuterionOS stays on the companion computer the entire time but needs to reflash the autopilot in order to revert.

hamishwillee replied:
Yay. That's what I said it should be :-)

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,72 @@
+
+## Building/Uploading Firmware
+
+Skynode PX4 `px4_fmu-v5x` binaries are built using the normal [developer environment](../dev_setup/dev_env.md) and are uploaded using either  `upload_skynode_usb` or `upload_skynode_wifi` upload targets.
```
```suggestion
Skynode PX4 `px4_fmu-v5x` binaries are built from source using the normal [developer environment](../dev_setup/dev_env.md) and [build commands](../dev_setup/building_px4.md), and are uploaded using either  `upload_skynode_usb` or `upload_skynode_wifi` upload targets.
```

MaEtUgR commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,66 @@
+
+To restore PX4:
+
+1. ?
```
This is the quickest way to revert I've found:
```
ssh auterion@10.41.1.1
cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar
tail --follow=name /shared_container_dir/fmu/update_status
```
Does this need to be in the script as well? I can see if that's doable.

hamishwillee replied:
I don't think this needs to be in the script - in particular because the locations of the dev image might not be in a standard location.

I'm going to copy this in, then add some questions for you.

MaEtUgR commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,66 @@
+
+Upstream PX4 will generally work, with the following caveats:
+
+- Configuration for your exact product may not be present.
```
```suggestion
- No officially supported release.
- Configuration for your exact product may not be present.
```
I think that's important to mention.

hamishwillee replied:
This is strongly implied by the "you can try this out on your development image". 

But if we must, should be at the end given the context. Proposed alternative text in https://github.com/PX4/PX4-user_guide/pull/2631/files#r1267464125

MaEtUgR replied:
I don't insist on that point but the reality is that not any random version flashed is/can be supported. So I thought it might be worth mentioning. You're right that it was already implied.

hamishwillee replied:
Ah, you mean "PX4 does not support this frame"/This is a manufacturer supported frame. Let me add that to the note below.

MaEtUgR reviewed (commented):
Nice! Thanks a lot @hamishwillee for estabilishing this page so quickly 🙏

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,66 @@
+
+## Skynode with Vanilla PX4
+
+Skynode comes with a managed release of PX4.
```
I don't want to use the words "Auterion PX4" - the website really only talks about Auterion OS now days, and I don't want to conflate Auterion and PX4 directly in docs on this site.

We'll try this.

```suggestion
Skynode comes with an Auterion-managed release of PX4.
```

We might also discuss this option if you don't like that:

> Skynode comes with a version of PX4 that has been optimised for use with Auterion products and services.

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,66 @@
+
+## Restoring the Default PX4 Firmware
+
+To restore PX4:
+
+1. ?
```
```suggestion
To restore the original Skynode development image you need to update the Auterion development PX4 .tar file to the mission computer, which will then upload it to the flight controller.

Assuming you have the Auterion tools installed your development computer:

1. SSH into Skynode.

   ```sh
   ssh auterion@10.41.1.1
   ```
   
   You will need to enter the default development login/passwords (auterion/auterion)

1. Copy the file `update-dev.tar` as shown.

   ```sh
   cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar
   ```

1. Skynode should automatically update the image on the flight controller part.
   You can monitor it for completion using the following command.

   ```sh
   tail --follow=name /shared_container_dir/fmu/update_status
   ```
```

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,66 @@
+
+- Configuration for your exact product may not be present.
+  You may lose the configuration for ESCs, batteries, sensor configuration, and so on.
+- Some parameters might have different default values in the PX4 release shipped with Auterion OS.
```
Again, not keen on the term "Auterion PX4" in this context. I probably would be OK with that on Auterion docs, assuming the branding still allows it (no mention on the website).

hamishwillee replied:
Setting resolved, though feel free to try up with a compromise.

MaEtUgR replied:
Also fine for me. Was just a suggestion to clarify the naming but I have no hard feelings.

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,82 @@
+- Configuration for your exact product may not be present.
+  You may lose the configuration for ESCs, batteries, sensor configuration, and so on.
+- Some parameters might have different default values in the PX4 release shipped with Auterion OS.
+- Features accessed by vendor-specific customisations running on the companion computer may not be present in PX4.
```
Added to the end "for discussion":

```suggestion
- Features accessed by vendor-specific customisations running on the companion computer may not be present in PX4.
- Auterion only provides support for Skynode running its own Auterion-managed version of PX4.
```

hamishwillee replied:
@MaEtUgR I think this covers the "not officially supported" even more - it makes clear that Auterion is the manufacturer and the circumstances under which they provide support.

hamishwillee commented on en/companion_computer/README.md:
```diff
@@ -24,6 +24,13 @@ The following boards are known to provide a good integration with PX4:
 
+## Managed Integrated Systems
+
+The following integrated companion computer/flight controller systems use managed versions of flight controller and companion computer software by default (unmanaged code can only run in a sandboxed environment on the companion computer).
```
Works for me. Thanks.

hamishwillee commented on en/companion_computer/README.md:
```diff
@@ -24,6 +24,13 @@ The following boards are known to provide a good integration with PX4:
 
+## Managed Integrated Systems
+
+The following integrated companion computer/flight controller systems use managed versions of flight controller and companion computer software by default.
```
```suggestion
The following integrated companion computer/flight controller systems use managed/custom versions of flight controller and companion computer software by default.
```

hamishwillee commented on en/companion_computer/auterion_skynode.md:
````diff
@@ -0,0 +1,83 @@
+   ```
+
+1. Skynode should automatically flash the new image to its flight controller.
+   You can monitor the update using the following command.
````
My guess is that this is why you are callling `tail` - so you can watch that Skynode does indeed install the flight controller tar automatically?

What should that look like/how will you know it is "done". Do you need to reboot or something afterwards?

MaEtUgR replied:
Yes, you're exactly right. Sorry for not mentioning explicitly 🙈 It's the exact same way the flashing script works. I tought about having an option in the script but how do you call it then? `make px4_fmu-v5x revert_skynode_usb` that would not be useful. It would need to be a separate script like: `./Tools/auterion/revert_skynode_px4.sh` or similar 🤔

hamishwillee commented on en/companion_computer/auterion_skynode.md:
````diff
@@ -0,0 +1,83 @@
+   ssh auterion@10.41.1.1
+   ```
+   
+   You will need to enter the default development login/password (auterion/auterion).
````
Assumption, SSH doesn't just let you in.

hamishwillee replied:
```suggestion
   You will need to enter the default development password when prompted (`auterion`).
```

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,83 @@
+   
+   You will need to enter the default development login/password (auterion/auterion).
+
+1. Copy the file `update.tar` onto Skynode (modify the source location to match the location of your Auterion images).
```
My assumption is that any user following any of this already knows how to get the image - is that true/reasonable? 

Essentially, please check that these steps are followable by a notional new user.

MaEtUgR replied:
```suggestion
```

That file is contained in AuterionOS, it's the PX4 version that shipped with the Auterion release that is currently running on the Skynode. If a user updates AuterionOS PX4 will be automatically flashed with the PX4 version that is baked in (even if he had a custom development one installed before). If he installs a custom development PX4 like we describe here then that is flashed and stays until the next AuterionOS update or until he flashes another one.

issuing the command
```
cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar
```
as auterion user sshed into the Skynode tells it: I have a new development version of PX4 and it is the PX4 version that shipped with AuterionOS. So it's reverting to the original PX4 version without the overhead of updating the entire AuterionOS.

hamishwillee replied:
@MaEtUgR  Ah, after SSH all commands are vehicle relative. I was misreading this as a copy from host computer. Thank you.

hamishwillee commented on assets/companion_computer/auterion_skynode/skynode_enterprise_tdw.png:
Yes, excellent idea

hamishwillee commented:
Needs a new review, particularly the last section. I have addressed most of your concerns, and thanks for that feedback!. I assume this won't be in PX4 v1.14?

MaEtUgR commented on en/companion_computer/auterion_skynode.md:
````diff
@@ -0,0 +1,84 @@
+   cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar
+   ```
+
+1. Skynode should automatically flash the new image to its flight controller.
+   You can monitor the update using the following command.
````
```suggestion
1. Skynode will automatically flash the original image to its flight controller.
   You can monitor the flashing process using the following command. It will not work anymore after the flashing is completed.
```

hamishwillee commented on en/companion_computer/auterion_skynode.md:
````diff
@@ -0,0 +1,86 @@
+   
+   You will need to enter the default development password when prompted (`auterion`).
+
+1. Copy/rename the original development image (named `update.tar`):
+
+   ```sh
+   cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar
+   ```
+
+1. Skynode will automatically flash the copied image to its flight controller.
+   You can monitor the flashing process using the following command.
+
+   ```sh
+   tail --follow=name /shared_container_dir/fmu/update_status
+   ```
+
+   After flashing completes Skynode will reboot.
````
@MaEtUgR This is what I think you mean then!

MaEtUgR commented on en/companion_computer/auterion_skynode.md:
````diff
@@ -0,0 +1,86 @@
+   ssh auterion@10.41.1.1
+   ```
+   
+   You will need to enter the default development password when prompted (`auterion`).
````
```suggestion
   You will need to enter the development password when prompted (by default `auterion`).
```

hamishwillee commented on en/companion_computer/auterion_skynode.md:
````diff
@@ -0,0 +1,86 @@
+   tail --follow=name /shared_container_dir/fmu/update_status
+   ```
+
+   After flashing completes Skynode will reboot.
````
It just doesn't sound quite right. But I think I understand what you mean.

```suggestion
   After flashing completes the `tail` command will no longer show progress (because the file is removed).
```

hamishwillee commented:
Cool. I think we might be good once the AutoPilot PR merges. Though it depends on whether this is going into PX4 v1.14.

MaEtUgR commented on en/companion_computer/auterion_skynode.md:
````diff
@@ -0,0 +1,86 @@
+
+## Restoring the Default PX4 Firmware
+
+To reinstall the original Skynode version of PX4:
+
+1. SSH into Skynode:
+
+   ```sh
+   ssh auterion@10.41.1.1
+   ```
+   
+   You will need to enter the development password when prompted (by default `auterion`).
+
+1. Copy/rename the original development image (named `update.tar`):
+
+   ```sh
+   cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar
+   ```
+
+1. Skynode will automatically flash the copied image to its flight controller.
+   You can monitor the flashing process using the following command.
+
+   ```sh
+   tail --follow=name /shared_container_dir/fmu/update_status
+   ```
+
+   After flashing completes the `tail` command will no longer show progress (because the file is removed).
````
```suggestion
To reinstall the original Skynode version of PX4 run the following command in the repository:
`./Tools/auterion/upload_skynode.sh --revert` and append an optional `--wifi` if you're connecting over wifi instead of USB

MaEtUgR replied:
I adjusted the script to support reverting, see https://github.com/PX4/PX4-Autopilot/pull/21842#issuecomment-1644403402

hamishwillee commented on en/companion_computer/auterion_skynode.md:
```diff
@@ -0,0 +1,74 @@
+
+## Restoring the Default PX4 Firmware
+
+To reinstall the original Skynode version of PX4 when connected via USB run the following command in the repository:
```
FYI only, I split out into two commands as code blocks render better (i.e. what you did was cool too). If there were more commands I'd do it like this, then explain each of the options:

```
./Tools/auterion/upload_skynode.sh --revert [--wifi]
```

MaEtUgR replied:
Makes sense.

MaEtUgR commented:
https://github.com/PX4/PX4-Autopilot/pull/21842 is merged, I also ported it to PX4 1.14 for the developer's convenience.
